### PR TITLE
Clean up stream_id representation

### PR DESCRIPTION
**Pull request by haaawk**
With https://github.com/scylladb/scylla/pull/5950 we changed the representation of stream_id in CDC Log from two int columns to a single blob column.

This PR cleans up stream_id representation internally. Now stream_id is stored as blob both in-memory and in internal CDC tables.

Tests: unit(dev)

### CODE REVIEW

**Comment by haaawk** [comment]
Release unit tests are passing as well.

File: cdc/generation.cc [hunk 1]
```diff
@@ -54,32 +54,37 @@ namespace cdc {
...
+    std::copy_n(_value.begin(), sizeof(int64_t), reinterpret_cast<int8_t *>(&res));
+    return res;
 }
 
 int64_t stream_id::second() const {
-    return _second;
+    assert(_value.size() == 2 * sizeof(int64_t));
+    int64_t res;
+    std::copy_n(_value.begin() + sizeof(int64_t), sizeof(int64_t), reinterpret_cast<int8_t *>(&res));
+    return net::ntoh(res);
```
Comment (piodul) [inline cdc/generation.cc]: `second` method performs endianness conversion, but `first` does not. Is it intended/correct?

Shouldn't we also byte-swap in the constructor `stream_id::stream_id(int64_t first, int64_t second)`?

Comment (haaawk) [reply cdc/generation.cc]: That's a leftover of my previous idea. Good catch. Thanks - will fix.

**Comment by haaawk** [comment]
I made the order of bytes to always be network order to make the code bullet proof. Thanks @piodul

**Comment by psarna** [comment]
> I made the order of bytes to always be network order to make the code bullet proof. Thanks @piodul

If it's 100% an internal decision and it's up to us, then little endian is way better. That's what Seastar RPC uses as well (source: https://github.com/scylladb/seastar/blob/master/doc/rpc.md), and the reason is that on x86, using the network order is pretty much "please always swap bytes each time I need to send or receive something". If there's a precedence in CDC and we already use big endian in there for some reason, then it's fine, otherwise I'd suggest to start with little endian, everyone's life will get easier.

**Comment by haaawk** [comment]
@psarna done - used little endian byte order

**Comment by haaawk** [comment]
Used write_le and read_le.

**Comment by psarna** [comment]
queued as c35160457

**Comment by haaawk** [comment]
Already merged. Closing...

**Pull request closed without merging**